server/routes/friendRouter.js

```diff
@@ -1,27 +1,29 @@
 const router = require('express').Router();
 const User = require('../models/userModel');
+const auth = require('../middleware/auth');
 
-router.get('/usersList', async (req, res) => {
+router.get('/', auth, async (req, res) => {
     try {
-        const user = await User.find({});
+        const users = await User.find({});
+        const filteredUsers = users.filter((user) => user._id != req.user);
```

Comment:
instead of filtering it manually, take advantage of your db queries to filter our the users

Reply:
Refactored with User.find({ '_id': { '$ne': req.user } })